lib/zhr_devs/submissions/task.ex

```diff
@@ -8,41 +8,100 @@ defmodule ZhrDevs.Submissions.Task do
 
   @dialyzer {:nowarn_function, [new: 0, new: 1, new: 2, new: 3]}
 
+  @supported_technologies :zhr_devs
+                          |> Application.compile_env(:supported_technologies)
+                          |> Enum.map(&Atom.to_string/1)
+
   import Algae
 
   @derive Jason.Encoder
 
+  defmodule Language do
+    @moduledoc "The supported language in which task is written"
+    @derive Jason.Encoder
+    defdata(Uptight.Text.t())
+
+    alias Uptight.Text, as: T
+
+    def new!(binary), do: %Language{language: T.new!(binary)}
+
+    def from_raw!(%Language{language: language}) do
```

Comment:
`from_raw!` takes the raw Elixir data which doesn't carry type information at all (here, of type `binary()`) and returns a completely wrapped structure (here, wraps in `Text` and `Language`).